### DISCUSSION TITLE
Cannot connect Jetson nano 4GB with Create 3

### DISCUSSION THREAD

**Discussion by siqiyan** [iRobotEducation/create3_docs#133] [Questions] [answered]
Hi, I have read the discussion and found similar problems, but none of them works for me so I want to ask here.

I got my new Create 3 and I follow the instructions to update the firmware to G.2.2, and set the RMW to `rmw_cyclonedds_cpp` already, and the robot can be teleoped normally using my laptop through wifi.

Then I am trying to use Jetson nano 4GB to connect to Create 3 via usb. My nano is connected directly to the usb C port on Create 3 without hub, and I use a LM2596 to convert the 14v battery power to 5v for powering the Jetson nano. I follow the [official docs](https://iroboteducation.github.io/create3_docs/setup/jetson/) to setup the Jetson nano but eventually I can only see `/parameter_events` and `/rosout` when typing `ros2 topic list`. The adapter board is switched to USB mode.

The strange thing is that I repeat the instruction multiple times and I was able to list full topics only once, but further `ros2 topic echo odom` didn't show anything.

I tried to `ping 192.168.186.2` and got 100% packet loss.

The output for `ifconfig l4tbr0` is:
```
l4tbr0: flags=4099<UP,BROADCAST,MULTICAST>  mtu 1500
        inet 192.168.186.3  netmask 255.255.255.0  broadcast 192.168.186.255
        inet6 fe80::1  prefixlen 128  scopeid 0x20<link>
        ether 76:fb:f0:b3:1d:95  txqueuelen 1000  (Ethernet)
        RX packets 0  bytes 0 (0.0 B)
        RX errors 0  dropped 0  overruns 0  frame 0
        TX packets 0  bytes 0 (0.0 B)
        TX errors 0  dropped 0 overruns 0  carrier 0  collisions 0
```

I also tied `dmesg | tail` on Jetson nano but I keep getting:
```
[  168.830320] IPv6: ADDRCONF(NETDEV_UP): l4tbr0: link is not ready
```

I thought maybe my cable is broken so I switched to another cable but nothing changed. Any help would be much appreciated!

**Comment by siqiyan** [comment, marked as answer]
Just an update, I can now list all topics using a USB type-C to Ethernet adapter between Jetson nano and Create 3, and set a static ip address to the `eth0` interface as specified in [here](https://github.com/iRobotEducation/create3_docs/discussions/59#discussioncomment-2780077), but it seems only `rmw_fastrtps_cpp` works for me.

**Reply by justinIRBT** [reply]
Did you setup a config file for cycloneDDS with multiple network interfaces as described here https://iroboteducation.github.io/create3_docs/setup/xml-config/
Also, is your create3 RMW set to cyclone or fastrtps on the create3 webserver?

**Reply by siqiyan** [reply]
Yes I tried that tutorial to config multiple network interfaces for cycloneDDS but not working. Currently I only use `eth0` on the Jetson nano for communicating ROS2, and the `wlan0` is for ssh only so I don't think configuring multiple interfaces is necessary, please correct me if I am wrong.

I did change the RMW_IMPLEMENTATION in the web interface accordingly, and click Save and Researt Application afterwards.
